File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 12 - Apprentice/ApprenticeDemo.C#/ApprenticeDemo/ApprenticeDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

using InventorApprentice;

namespace ApprenticeDemo
{
    public partial class Form1 : Form
    {
        ApprenticeServerComponent mApprenticeServer ; 
        ApprenticeServerDocument mCurrentDoc ; 
        ApprenticeServerDrawingDocument mCurrentDrawingDoc;

        public Form1()
        {
            InitializeComponent();
        }

        private void OpenButton_Click(object sender, EventArgs e)
        {
            if (((mCurrentDoc != null)))
            {
                mCurrentDoc.Close();
            }

            string filename = OpenFile();


            if ((!string.IsNullOrEmpty(filename)))
            {
                mCurrentDoc = mApprenticeServer.Open(filename);


                if ((mCurrentDoc.DocumentType == DocumentTypeEnum.kDrawingDocumentObject))
                {
                    mCurrentDrawingDoc = (ApprenticeServerDrawingDocument)mCurrentDoc;
                }


                IPictureDisp oPicDisp = (IPictureDisp)mCurrentDoc.Thumbnail;
                Image image = Microsoft.VisualBasic.Compatibility.VB6.Support.IPictureDispToImage(oPicDisp);

                //image.Save("c:\Temp\Thumbnail.bmp", System.Drawing.Imaging.ImageFormat.Bmp)

                PreviewPic.Image = image;
                PreviewPic.Refresh();

                lbFilename.Text = "File: " + mCurrentDoc.DisplayName;

                ViewerButton.Enabled = true;
            }


        }

        private void ViewerButton_Click(object sender, EventArgs e)
        {
            Form2 viewer = new Form2(mCurrentDoc.FullFileName); 
            viewer.ShowDialog();
        }

        private void btnSetPro_Click(object sender, EventArgs e)
        {
            SetProperty(TextBoxAuthor.Text);
        }

        //Small helper function that prompts user for a file selection
        private string OpenFile()
        {

            string filename = "";

            System.Windows.Forms.OpenFileDialog ofDlg = new System.Windows.Forms.OpenFileDialog();

            string user = System.Windows.Forms.SystemInformation.UserName;

            ofDlg.Title = "Open Inventor File";
            ofDlg.InitialDirectory = "C:\\Documents and Settings\\" + user + "\\Desktop\\";

            //"Inventor Files (*.ipt)|*.ipt|Inventor Assemblies (*.iam)|*.iam|Inventor Drawings (*.idw)|*.idw"
            ofDlg.Filter = "Inventor files (*.ipt; *.iam; *.idw)|*.ipt;*.iam;*.idw";
            ofDlg.FilterIndex = 1;
            ofDlg.RestoreDirectory = true;

            if ((ofDlg.ShowDialog() == DialogResult.OK))
            {
                filename = ofDlg.FileName;
            }

            return filename;

        }

        // set property [Author]
        private void SetProperty(string author)
        {
            ApprenticeServerDocument oApprenticeDoc = default(ApprenticeServerDocument);
            oApprenticeDoc = mApprenticeServer.Open(OpenFile());

            //Get "Inventor Summary Information" PropertySet
            PropertySet oPropertySet = oApprenticeDoc.PropertySets["{F29F85E0-4FF9-1068-AB91-08002B27B3D9}"];

            //Get Author property
            InventorApprentice.Property oProperty = oPropertySet["Author"];

            oProperty.Value = author;

            oApprenticeDoc.PropertySets.FlushToFile();
            oApprenticeDoc.Close();

        }


        private void Form1_Load(object sender, EventArgs e)
        {
               mApprenticeServer = new ApprenticeServerComponent();
            ViewerButton.Enabled = false;

            PreviewPic.SizeMode = PictureBoxSizeMode.StretchImage;

            mCurrentDoc = null;

        }


    }
}

File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 12 - Apprentice/ApprenticeDemo.VB.Net/ApprenticeDemo/ApprenticeDemo/Form1.vb

Imports InventorApprentice

Public Class Form1

    Dim mApprenticeServer As ApprenticeServerComponent
    Dim mCurrentDoc As ApprenticeServerDocument
    Dim mCurrentDrawingDoc As ApprenticeServerDrawingDocument

    Private Sub OpenButton_Click(sender As System.Object, e As System.EventArgs) Handles OpenButton.Click
        If (Not mCurrentDoc Is Nothing) Then
            mCurrentDoc.Close()
        End If

        Dim filename As String = OpenFile()

        If (filename <> "") Then

            mCurrentDoc = mApprenticeServer.Open(filename)

            If (mCurrentDoc.DocumentType = DocumentTypeEnum.kDrawingDocumentObject) Then

                mCurrentDrawingDoc = mCurrentDoc

            End If


            Dim oPicDisp As IPictureDisp = mCurrentDoc.Thumbnail
            Dim image As Image = Microsoft.VisualBasic.Compatibility.VB6.IPictureDispToImage(oPicDisp)

            'image.Save("c:\Temp\Thumbnail.bmp", System.Drawing.Imaging.ImageFormat.Bmp)

            PreviewPic.Image = image
            PreviewPic.Refresh()

            lbFilename.Text = "File: " + mCurrentDoc.DisplayName

            ViewerButton.Enabled = True
        End If
    End Sub

    Private Sub ViewerButton_Click(sender As System.Object, e As System.EventArgs) Handles ViewerButton.Click
        Dim viewer As Form2 = New Form2(mCurrentDoc.FullFileName)
        viewer.ShowDialog()
    End Sub

    'Small helper function that prompts user for a file selection
    Private Function OpenFile() As String

        Dim filename As String = ""

        Dim ofDlg As System.Windows.Forms.OpenFileDialog = New System.Windows.Forms.OpenFileDialog()

        Dim user As String = System.Windows.Forms.SystemInformation.UserName

        ofDlg.Title = "Open Inventor File"
        ofDlg.InitialDirectory = "C:\Documents and Settings\" + user + "\Desktop\"

        '"Inventor Files (*.ipt)|*.ipt|Inventor Assemblies (*.iam)|*.iam|Inventor Drawings (*.idw)|*.idw"
        ofDlg.Filter = "Inventor files (*.ipt; *.iam; *.idw)|*.ipt;*.iam;*.idw"
        ofDlg.FilterIndex = 1
        ofDlg.RestoreDirectory = True

        If (ofDlg.ShowDialog() = DialogResult.OK) Then
            filename = ofDlg.FileName
        End If

        OpenFile = filename

    End Function


    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        mApprenticeServer = New ApprenticeServerComponent()
        ViewerButton.Enabled = False

        Me.PreviewPic.SizeMode = PictureBoxSizeMode.StretchImage

        mCurrentDoc = Nothing

    End Sub

    Private Sub btnSetPro_Click(sender As System.Object, e As System.EventArgs) Handles btnSetPro.Click
        SetProperty(TextBoxAuthor.Text)
    End Sub

    Private Sub SetProperty(ByVal author As String)

        Dim oApprenticeDoc As ApprenticeServerDocument
        oApprenticeDoc = mApprenticeServer.Open(OpenFile())

        'Get "Inventor Summary Information" PropertySet
        Dim oPropertySet As PropertySet = oApprenticeDoc.PropertySets("{F29F85E0-4FF9-1068-AB91-08002B27B3D9}")

        'Get Author property
        Dim oProperty As InventorApprentice.Property = oPropertySet.Item("Author")

        oProperty.Value = author

        oApprenticeDoc.PropertySets.FlushToFile()
        oApprenticeDoc.Close()

    End Sub
End Class
